packages/client/package.json

```diff
@@ -15,7 +15,7 @@
         "typescript": "~4.6.2"
     },
     "engines": {
-        "node": ">=12"
+        "node": ">=18"
```

Comment:
I think Node 18 is ok to declare as a minimum engine, since Node 16 maintenance LTS is about to end at the end of this month. See [Node.js Releases](https://nodejs.dev/en/about/releases/). `@blueprintjs/node-build-scripts` already declares a minimum Node version of 18.13.